src/pybind/mgr/dashboard/frontend/src/app/ceph/block/iscsi/iscsi.component.spec.ts

```diff
@@ -39,10 +42,88 @@ describe('IscsiComponent', () => {
   beforeEach(() => {
     fixture = TestBed.createComponent(IscsiComponent);
     component = fixture.componentInstance;
+    tcmuIscsiService = TestBed.get(TcmuIscsiService);
     fixture.detectChanges();
   });
 
   it('should create', () => {
     expect(component).toBeTruthy();
   });
+
+  it('should refresh without stats available', () => {
+    const tcmuiscsiData = {
+      daemons: [
+        {
+          server_hostname: 'node1.ceph.local',
+          version: 'ceph version 13.2.2 (02899bfda814146b021136e9d8e80eba494e1126) mimic (stable)',
+          optimized_paths: 0,
+          non_optimized_paths: 1
+        },
+        {
+          server_hostname: 'node2.ceph.local',
+          version: 'ceph version 13.2.2 (02899bfda814146b021136e9d8e80eba494e1126) mimic (stable)',
+          optimized_paths: 0,
+          non_optimized_paths: 1
+        }
+      ],
+      images: [
+        {
+          device_id: 'rbd/disk_1',
+          pool_name: 'rbd',
+          name: 'disk_1',
+          id: '11ce6b8b4567',
+          optimized_paths: [],
+          non_optimized_paths: ['node1.ceph.local', 'node2.ceph.local']
+        }
+      ]
+    };
+    spyOn(tcmuIscsiService, 'tcmuiscsi').and.callFake(() => of(tcmuiscsiData));
+    component.refresh();
+    expect(component.images[0].stats).toBeUndefined();
+    expect(component.images[0].stats_history).toBeUndefined();
```

Comment:
If I remove the if clause this test will also succeed.

Reply:
You can expect that `cdIsBinary` is true to verify the right behavior, as this will than fail if the if clause isn't there.